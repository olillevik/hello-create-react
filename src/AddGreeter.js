import React, {Component} from 'react';
import './AddGreeter.css';

class AddGreeter extends Component {

    constructor(props) {
        super(props);
        this.state = {greetingName: ''}
        this.handleUpdate = this.handleUpdate.bind(this);
        this.addGreeting = this.addGreeting.bind(this);
    }

    render() {
        return (
            <div className='AddGreeter'>
                <input type='text' onChange={this.handleUpdate}/>
                &nbsp;&nbsp;
                <button onClick={this.addGreeting}>add</button>
            </div>
        );
    }

    handleUpdate(event) {
        this.setState({greetingName: event.target.value});
    }

    addGreeting() {
        this.props.addGreeting(this.state.greetingName);
        this.setState({greetingName: ''});
    }
}

export default AddGreeter;