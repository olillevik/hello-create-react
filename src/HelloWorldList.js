import React, {Component} from 'react';
import './HelloWorldList.css';

import HelloWorld from './HelloWorld';
import AddGreeter from './AddGreeter';

class HelloWorldList extends Component {

    constructor(props) {
        super(props);
        this.state = {greetings: ['Jim', 'Sally', 'Bender']};
        this.addGreetingToList = this.addGreetingToList.bind(this);
        this.removeGreetingFromList = this.removeGreetingFromList.bind(this);
    }

    render() {
        return (
            <div className='HelloWorldList'>
                <AddGreeter addGreeting={this.addGreetingToList}/>
                {this.renderGreetings()}

            </div>
        );
    }

    renderGreetings() {
        return this.state.greetings.map(name => (
            <HelloWorld key={name} name={name} removeGreeting={this.removeGreetingFromList}/>
        ));
    }

    addGreetingToList(newName) {
        this.setState({greetings: [...this.state.greetings, newName]});
    }

    removeGreetingFromList(removeName) {
        const filteredGreetings = this.state.greetings.filter(name => {
                return name != removeName;
            }
        );
        this.setState({greetings: filteredGreetings});
    }
}

export default HelloWorldList;