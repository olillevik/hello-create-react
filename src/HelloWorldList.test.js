import React from 'react';
import {shallow} from 'enzyme';
import renderer from 'react-test-renderer';

import HelloWorldList from './HelloWorldList';
import HelloWorld from './HelloWorld'
import AddGreeter from './AddGreeter'

describe(HelloWorldList, () => {
    const component = shallow(
        <HelloWorldList/>
    );

    it('renders and matches snapshot', () => {
        const component1 = renderer.create(
            <HelloWorldList/>
        )
        const tree = component1.toJSON();
        expect(tree).toMatchSnapshot();
    })

    it('contains AddGreeter component', () => {
        expect(component.find(AddGreeter)).toHaveLength(1);
    });

    it('contains the same number if HelloWorld components as greetings', () => {
        expect(component.find(HelloWorld)).toHaveLength(component.state('greetings').length);
    });

    it('adds a greeting when the add greeting function is called', () => {
        const prevLength = component.find(HelloWorld).length;
        component.instance().addGreetingToList('Sample');
        component.update();
        const newLength = component.find(HelloWorld).length;
        expect(prevLength).toEqual(newLength - 1);
    });

    it('removes a greeeting when the remove greeting function is called', () => {
        const prevLength = component.find(HelloWorld).length;
        component.instance().removeGreetingFromList(component.state('greetings')[0]);
        component.update();
        const newLength = component.find(HelloWorld).length;
        expect(prevLength).toEqual(newLength + 1);
    });
});