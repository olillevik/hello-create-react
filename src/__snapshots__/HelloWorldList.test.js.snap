// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`class HelloWorldList extends _react.Component {

    constructor(props) {
        super(props);
        this.state = { greetings: ['Jim', 'Sally', 'Bender'] };
        this.addGreetingToList = this.addGreetingToList.bind(this);
        this.removeGreetingFromList = this.removeGreetingFromList.bind(this);
    }

    render() {
        return (
            _react2.default.createElement('div', { className: 'HelloWorldList', __source: { fileName: _jsxFileName, lineNumber: 18 }, __self: this },
                _react2.default.createElement(_AddGreeter2.default, { addGreeting: this.addGreetingToList, __source: { fileName: _jsxFileName, lineNumber: 19 }, __self: this }),
                this.renderGreetings()));



    }

    renderGreetings() {
        return this.state.greetings.map(name =>
        _react2.default.createElement(_HelloWorld2.default, { key: name, name: name, removeGreeting: this.removeGreetingFromList, __source: { fileName: _jsxFileName, lineNumber: 28 }, __self: this }));

    }

    addGreetingToList(newName) {
        this.setState({ greetings: [...this.state.greetings, newName] });
    }

    removeGreetingFromList(removeName) {
        const filteredGreetings = this.state.greetings.filter(name => {
            return name != removeName;
        });

        this.setState({ greetings: filteredGreetings });
    }} renders and matches snapshot 1`] = `
<div
  className="HelloWorldList"
>
  <div
    className="AddGreeter"
  >
    <input
      onChange={[Function]}
      type="text"
    />
      
    <button
      onClick={[Function]}
    >
      add
    </button>
  </div>
  <div
    className="HelloWorld"
  >
    Hello
     
    Jim
    !
    <br />
    <button
      className="frenchify"
      onClick={[Function]}
    >
      Frenchify!
    </button>
    <br />
    <button
      className="remove"
      onClick={[Function]}
    >
      Remove me!
    </button>
  </div>
  <div
    className="HelloWorld"
  >
    Hello
     
    Sally
    !
    <br />
    <button
      className="frenchify"
      onClick={[Function]}
    >
      Frenchify!
    </button>
    <br />
    <button
      className="remove"
      onClick={[Function]}
    >
      Remove me!
    </button>
  </div>
  <div
    className="HelloWorld"
  >
    Hello
     
    Bender
    !
    <br />
    <button
      className="frenchify"
      onClick={[Function]}
    >
      Frenchify!
    </button>
    <br />
    <button
      className="remove"
      onClick={[Function]}
    >
      Remove me!
    </button>
  </div>
</div>
`;
